pkg/storage/stores/shipper/compactor/retention/expiration.go

```diff
@@ -119,17 +132,19 @@ Outer:
 	return globalRetention
 }
 
-// findLatestRetentionStartTime returns the latest retention start time overall and by each user.
-func findLatestRetentionStartTime(now model.Time, limits Limits) (model.Time, map[string]model.Time) {
+// findLatestRetentionStartTime returns the latest retention start time overall, just default config and by each user.
+func findLatestRetentionStartTime(now model.Time, limits Limits) (model.Time, model.Time, map[string]model.Time) {
 	// find the smallest retention period from default limits
 	defaultLimits := limits.DefaultLimits()
-	smallestRetentionPeriod := defaultLimits.RetentionPeriod
+	smallestDefaultRetentionPeriod := defaultLimits.RetentionPeriod
 	for _, streamRetention := range defaultLimits.StreamRetention {
```

Comment:
We don use default stream retention right ? We support it though.

Reply:
yeah, we don't use it in any of our envs but we do have a support for it.